File: src/problem_2/problem_2.c
#include <stdio.h>

/*
 * Solution to problem 2
 *
 * https://projecteuler.net/problem=2
 */

int main(void) {
    int solution = fibonacci();
    printf("%i", solution);
}

int fibonacci(void) {
    int limit = 4000000;
    int a = 1;
    int b = 2;
    int sum = 0;
    int c = 0;

    while (b < limit) {
        if (b % 2 == 0) {
            sum = sum + b;
        }
        c = a + b;
        a = b;
        b = c;
    }

    return sum;
}

File: src/problem_2/problem_2.py
# Solution to problem 2
#
# https://projecteuler.net/problem=2


def fibonacci():
    limit = 4000000
    a = 1
    b = 2
    s = 0

    while b < limit:
        if b % 2 == 0:
            s += b
        c = a + b
        a = b
        b = c

    return s

print(fibonacci())

